runtime/src/main/java/com/flipkart/flux/controller/WorkFlowExecutionController.java

```diff
@@ -260,8 +263,9 @@ private StateMachine retrieveStateMachine(Long stateMachineInstanceId) {
     public void redriveTask(Long taskId) {
         State state = statesDAO.findById(taskId);
         if(state != null && isTaskRedrivable(state.getStatus()) && state.getAttemptedNoOfRetries() < state.getRetryCount()) {
+            StateMachine stateMachine = retrieveStateMachine(state.getStateMachineId());
```

Comment:
we don't have state machine id, unlike earlier case, we derive that from the state